modules/entity_version_history/tests/src/FunctionalJavascript/HistoryOverviewTest.php

```diff
@@ -0,0 +1,199 @@
+<?php
+
+declare(strict_types = 1);
+
+namespace Drupal\Tests\entity_version_history\Functional;
+
+use Drupal\entity_version_history_test\EventSubscriber\TestHistoryOverviewAlterEventSubscriber;
+use Drupal\FunctionalJavascriptTests\WebDriverTestBase;
+use Drupal\node\Entity\Node;
+
+/**
+ * Tests the history overview table.
+ *
+ * @group entity_version_history
+ */
+class HistoryOverviewTest extends WebDriverTestBase {
+
+  /**
+   * An array of node revisions.
+   *
+   * @var \Drupal\node\NodeInterface[]
+   */
+  protected $nodes;
+
+  /**
+   * A user with administrative permissions.
+   *
+   * @var \Drupal\user\UserInterface
+   */
+  protected $adminUser;
+
+  /**
+   * {@inheritdoc}
+   */
+  protected static $modules = [
+    'node',
+    'entity_version',
+    'entity_version_history',
+    'entity_version_history_test',
+  ];
+
+  /**
+   * {@inheritdoc}
+   */
+  protected function setUp() {
+    parent::setUp();
+
+    // Create user.
+    $this->adminUser = $this->drupalCreateUser([
+      'access entity version history configuration',
+      'access administration pages',
+      'access entity version history',
+      'edit any first_bundle content',
+      'administer content types',
+    ]);
+    $this->drupalLogin($this->adminUser);
+
+    // Create a history tab setting for the corresponding entity type
+    // and bundle.
+    $entity_type_manager = $this->container->get('entity_type.manager');
+    $history_storage = $entity_type_manager->getStorage('entity_version_history_settings');
+    $history_storage->create([
+      'target_entity_type_id' => 'node',
+      'target_bundle' => 'first_bundle',
+      'target_field' => 'field_entity_version',
+    ])->save();
+
+    // We need to invalidate cache because in our route subscriber and our
+    // entity type alter we are depending on the config we created above.
+    $entity_type_manager->clearCachedDefinitions();
```

Comment:
M...isn't this a sign that we need to that in the actual code? At the moment you do it in the form but maybe it needs to be moved to an entity update hook or config subscriber event no?

Reply:
Right, once an entity is created/changed these cache needs to be cleared. I will look into those two you mentioned.